src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -218,19 +227,32 @@ public synchronized void stop() {
 
     MongoCursor<BsonDocument> createCursor(final MongoSourceConfig cfg, final MongoClient mongoClient) {
         LOGGER.debug("Creating a MongoCursor");
-        ChangeStreamIterable<Document> changeStream = getChangeStreamIterable(cfg, mongoClient);
-        BsonDocument resumeAfter = getResumeAfter(cfg);
-        if (resumeAfter != null) {
-            LOGGER.info("Resuming the change stream at the previous offset");
-            changeStream.resumeAfter(resumeAfter);
-            cachedResumeAfter = null;
+        BsonDocument resumeToken = getResumeToken(cfg);
+        return tryCreateCursor(cfg, mongoClient, resumeToken)
+                .orElseGet(() -> resumeToken == null ? null : tryCreateCursor(cfg, mongoClient, null).orElse(null));
```

Comment:
If the first creation of the cursor fails and there was a resumeToken- eg the was a command exception (doesn't support startAfter) then create a new change stream cursor from now.